src/main/java/org/opensearch/neuralsearch/search/HybridTopScoreDocCollector.java

```diff
@@ -47,16 +46,35 @@ public HybridTopScoreDocCollector(int numHits, HitsThresholdChecker hitsThreshol
     }
 
     @Override
-    public LeafCollector getLeafCollector(LeafReaderContext context) throws IOException {
+    public LeafCollector getLeafCollector(LeafReaderContext context) {
         docBase = context.docBase;
 
-        return new TopScoreDocCollector.ScorerLeafCollector() {
+        return new LeafCollector() {
             HybridQueryScorer compoundQueryScorer;
 
             @Override
             public void setScorer(Scorable scorer) throws IOException {
-                super.setScorer(scorer);
-                compoundQueryScorer = (HybridQueryScorer) scorer;
+                if (scorer instanceof HybridQueryScorer) {
+                    compoundQueryScorer = (HybridQueryScorer) scorer;
+                } else {
+                    compoundQueryScorer = getHybridQueryScorer(scorer);
+                }
+            }
+
+            private HybridQueryScorer getHybridQueryScorer(final Scorable scorer) throws IOException {
+                if (scorer == null) {
+                    return null;
+                }
+                if (scorer instanceof HybridQueryScorer) {
+                    return (HybridQueryScorer) scorer;
+                }
+                for (Scorable.ChildScorable childScorable : scorer.getChildren()) {
+                    HybridQueryScorer hybridQueryScorer = getHybridQueryScorer(childScorable.child);
+                    if (hybridQueryScorer != null) {
+                        return hybridQueryScorer;
+                    }
+                }
+                return null;
```

Comment:
this will definitely cause a NPE in that line. Let me add a code where we throw a runtime exception in this case, as we cannot proceed with score collections if all scorers are empty/null.

Reply:
Lets add enough logs to ensure that we understand why the scorer was null.

On the other side do we know in which cases the scorer can be null? and is it a valid case?

Reply:
I don't have info on a cases when `scorer` can be null and it's a valid scenario. I'll add more logs